module/icp/algs/aes/aes_impl.c

```diff
@@ -1502,43 +153,36 @@ int
 aes_decrypt_block(const void *ks, const uint8_t *ct, uint8_t *pt)
 {
 	aes_key_t	*ksch = (aes_key_t *)ks;
+	const aes_impl_ops_t	*ops = aes_impl_get_ops();
```

Comment:
why is it that here we grab a new ops, when in `aes_encrypt()` we grab the one in the ksch? I'm worried about this because each encryption / decryption operation should use the same implementation from beginning to end.

Reply:
this is a mistake on my part, thanks for catching that.